### PR TITLE
keyboard notifications handling

### PR DESCRIPTION
**Pull request by petropavel13**
example of usage https://gist.github.com/petropavel13/a9a387b674cea7d0ed1205caffcdb898

### CODE REVIEW

File: LeadKit/LeadKit/Extensions/UIViewAnimationCurve/UIViewAnimationCurve+AnimationOptions.swift [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+//  LeadKit
+//
+//  Created by Ivan Smolin on 12/12/16.
+//  Copyright © 2016 Touch Instinct. All rights reserved.
+//
+
+import Foundation
+
+public extension UIViewAnimationCurve {
+
```
Comment (iON1k) [inline LeadKit/LeadKit/Extensions/UIViewAnimationCurve/UIViewAnimationCurve+AnimationOptions.swift]: For what this extension? I didn't find using of this

File: LeadKit/LeadKit/Extensions/Notification/Notification+UIKeyboardInfoValues.swift [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+//
+
+import Foundation
+
+public extension Notification {
+
+    /// CGRect value for UIKeyboardFrameBeginUserInfoKey or nil
+    var keyboardFrameBegin: CGRect? {
+        return (userInfo?[UIKeyboardFrameBeginUserInfoKey] as? NSValue)?.cgRectValue
+    }
```
Comment (iON1k) [inline LeadKit/LeadKit/Extensions/Notification/Notification+UIKeyboardInfoValues.swift]: I don't think that any notifications should have these properties.
Maybe move them to helper class?

File: LeadKit/LeadKit/Extensions/UIViewAnimationCurve/UIViewAnimationCurve+AnimationOptions.swift (hunk 1 above)
Comment (petropavel13) [inline LeadKit/LeadKit/Extensions/UIViewAnimationCurve/UIViewAnimationCurve+AnimationOptions.swift]: UIView.animate(withDuration: values.animationDuration,
                       delay: 0.0,
                       options: values.animationCurve.animationOptions,

File: LeadKit/LeadKit/Extensions/Notification/Notification+UIKeyboardInfoValues.swift (hunk 2 above)
Comment (petropavel13) [inline LeadKit/LeadKit/Extensions/Notification/Notification+UIKeyboardInfoValues.swift]: did you mean `KeyboardDidNotificationValues` and `KeyboardWillNotificationValues`?

Comment (iON1k) [inline LeadKit/LeadKit/Extensions/Notification/Notification+UIKeyboardInfoValues.swift]: Yes, something like this.
But for avoid code duplicates in KeyboardDidNotificationValues and KeyboardWillNotificationValues maybe move it to another class

**Pull request merged**